FileShareService.DesktopClient.Core/JobsParser.cs

```diff
@@ -26,12 +35,144 @@ public Jobs.Jobs Parse(string jobs)
             );
             try
             {
-                return JsonConvert.DeserializeObject<Jobs.Jobs>(jobs, jsonSerializerSettings);
+                JToken jobsToken = JToken.Parse(jobs);
+
+                var batchJobs = jobsToken.SelectToken("jobs");
+
+                if (batchJobs == null || batchJobs.Type != JTokenType.Array || !batchJobs.HasValues)
+                {
+                    throw new JsonReaderException("Configuration file formatted incorrectly. Unable to find a job to process.");
+                }
+
+                List<IJob> jobCollection = new List<IJob>();
+
+                jobIdCollection = new List<string>();
+
+                ErrorDeserializingJobsJob? errorJob = null;
+
+                // Deserialize each job and validate
+                foreach (JToken batchJob in batchJobs)
+                {
+                    string jobAction = string.Empty;
+
+                    List<string> errors = ValidateJobActionAndDisplayName(batchJob, out jobAction);
+
+                    if(errors.Any())
+                    {
+                        if(errorJob == null)
+                        {
+                            errorJob = new ErrorDeserializingJobsJob();
+                        }
+                        errorJob.ErrorMessages.AddRange(errors);
+
+                        continue;
+                    }
+
+                    IJob? job = null;
+                    string jsonString = Convert.ToString(batchJob);
+
+                    switch (jobAction)
```

Comment:
Please convert the switch statement to a more concise switch expression
````
                    IJob? job = jobAction switch {
                        NewBatchJob.JobAction => JsonConvert.DeserializeObject<NewBatchJob>(jsonString, jsonSerializerSettings),
                        AppendAclJob.JobAction => JsonConvert.DeserializeObject<AppendAclJob>(jsonString, jsonSerializerSettings),
                        SetExpiryDateJob.JobAction => JsonConvert.DeserializeObject<SetExpiryDateJob>(jsonString, jsonSerializerSettings),
                        _ => null
                    };
````

Reply:
Modified as suggested.